tests/vuefire.spec.js

```diff
@@ -25,6 +27,36 @@ describe('VueFire', function () {
     })
   })
 
+  describe('is callable as Function', function () {
+    it('returns correct ref on function call', function (done) {
+      var ChildComponent = Vue.extend({
+        name: 'ChildComponent',
+        firebase: function () {
+          expect(this.$root.database).to.deep.equal(firebaseDb)
+          done()
+        },
+        template: '<div>test</div>'
+      })
+      new Vue({
+        data: function () {
+          return {
+            database: firebaseDb
+          }
+        },
+        ready: function () {
+          expect('poop').to.equal('notpoop')
+        },
+        components: {
+          'child-component': ChildComponent
+        },
+        template: '<div><child-component></child-component></div>'
+      }).$mount()
+      Vue.nextTick(function () {
```

Comment:
Damn, I seem to not have removed my initial attempts at getting the test to work. I'll fix that ASAP.